EventSource/EventSource.m

```diff
@@ -196,7 +196,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data
                 });
             }
             
-            if (e.event != nil) {
+            if (e.event != nil && ![e.event isEqualToString:MessageEvent]) {
```

Comment:
This is redundant, but I'll clarify why. `MessageEvent` handlers receive **all** events—that is, named & unnamed events. If the server sends an event named `message`, that's not a message event, but a named event called `message`.  Here's an example: 

```objc
EventSource *source = [EventSource eventSourceWithURL:[NSURL URLWithString:@"http://127.0.0.1:8000/"]];
[source onMessage:^(Event *event) {
    // This will receive all events regardless of whether it's a named or unnamed event
}];

[source addEventListener:@"message" handler:^(Event *e) {
    // This will only ever receive events with the name 'message'.
}];
```